napari/layers/graph/graph.py

```diff
@@ -0,0 +1,339 @@
+
+except ModuleNotFoundError:
+    BaseGraph = None
+    UndirectedGraph = None
+
+
+class Graph(_BasePoints):
+    def __init__(
+        self,
+        data=None,
+        *,
+        ndim=None,
+        features=None,
+        feature_defaults=None,
+        text=None,
+        symbol='o',
+        size=10,
+        border_width=0.05,
+        border_width_is_relative=True,
+        border_color='dimgray',
+        border_color_cycle=None,
+        border_colormap='viridis',
+        border_contrast_limits=None,
+        face_color='white',
+        face_color_cycle=None,
+        face_colormap='viridis',
+        face_contrast_limits=None,
+        out_of_slice_display=False,
+        n_dimensional=None,
+        name=None,
+        metadata=None,
+        scale=None,
+        translate=None,
+        rotate=None,
+        shear=None,
+        affine=None,
+        opacity=1,
+        blending='translucent',
+        visible=True,
+        cache=True,
+        experimental_clipping_planes=None,
+        shading='none',
+        canvas_size_limits=(2, 10000),
+        antialiasing=1,
+        shown=True,
+    ) -> None:
+        if BaseGraph is None:
+            raise RuntimeError(
+                trans._(
+                    "`napari-graph` module is required by the graph layer."
+                )
+            )
+
+        self._data = self._fix_data(data, ndim)
+        self._edges_indices_view: ArrayLike = []
+
+        super().__init__(
+            self._data,
+            ndim=self._data.ndim,
+            features=features,
+            feature_defaults=feature_defaults,
+            text=text,
+            symbol=symbol,
+            size=size,
+            border_width=border_width,
+            border_width_is_relative=border_width_is_relative,
+            border_color=border_color,
+            border_color_cycle=border_color_cycle,
+            border_colormap=border_colormap,
+            border_contrast_limits=border_contrast_limits,
+            face_color=face_color,
+            face_color_cycle=face_color_cycle,
+            face_colormap=face_colormap,
+            face_contrast_limits=face_contrast_limits,
+            out_of_slice_display=out_of_slice_display,
+            n_dimensional=n_dimensional,
+            name=name,
+            metadata=metadata,
+            scale=scale,
+            translate=translate,
+            rotate=rotate,
+            shear=shear,
+            affine=affine,
+            opacity=opacity,
+            blending=blending,
+            visible=visible,
+            cache=cache,
+            experimental_clipping_planes=experimental_clipping_planes,
+            shading=shading,
+            canvas_size_limits=canvas_size_limits,
+            antialiasing=antialiasing,
+            shown=shown,
+        )
+
+        # TODO:
+        # dummy events because of VispyGraphLayer's VispyPointsLayerinheritance
+        # should be removed in 6.0.0
```

Comment:
```suggestion
        # should be removed in 0.6.0
```

probably a bit earlier? 😂